src/clojure/nrepl/server.clj

```diff
@@ -34,16 +55,76 @@
     (catch java.io.IOException e
       (log e "Failed to close " x))))
 
+(defn- default-bytes-logger
+  [channel ba]
+  (log/info channel (map byte ba)))
+
+(defn- default-string-logger
+  [channel ba]
+  (log/info channel (str/join (map char ba))))
```

Comment:
Don't you need the encoding scheme of the transport to try to get some meaningful string out of this?

Reply:
You don't need to use the transport as the current transports have the bytes representing chars, it just sniffs the socket independent of the transport, we certainly need comments of other people here to see the usefulness of it or if they need other ways to do it.